### PR TITLE
Add flow type to disco store

### LINKED ISSUE #6279: Add flow types to Disco store
Supports #2917 

---

In order to make the disco pane code easier to update (and more readable), we should have more flow coverage. This issue is about adding flow types to the disco store.

### PR DESCRIPTION
**Pull request by willdurand**
Fix #6279

---

It is a small step for a better disco app.

### CODE REVIEW

**Comment by kumar303** [comment]
Awesome! Is there an issue for this? Since we don't use PRs themselves for tracking I think it always helps to add an issue (even just a one-liner).

**Comment by willdurand** [comment]
> Awesome! Is there an issue for this? Since we don't use PRs themselves for tracking I think it always helps to add an issue (even just a one-liner).

Ah right sorry, yes I filed one for this and will open new ones for the other bits.

File: src/core/types/store.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { ReactRouterHistoryType } from 'core/types/router';
+
+// This is a type function that takes a state type and returns a reducer
+// type, i.e. a function that accepts and returns the same state type.
+/* eslint-disable no-undef */
```
Comment (kumar303) [inline src/core/types/store.js]: You could try removing this to see what happens. It's some kind of eslint/flow bug.

Comment (willdurand) [reply src/core/types/store.js]: it does not work :( I tried and it complains about `AnyState` not being defined..

**Pull request merged**